File: src/cn/vove7/plugin/rest/filetype/RestFile.java
package cn.vove7.plugin.rest.filetype;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;
import cn.vove7.plugin.rest.psi.RestESeparator;
import cn.vove7.plugin.rest.psi.RestRequest;

/**
 *
 */
public class RestFile extends PsiFileBase {

    protected RestFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, RestLanguage.INSTANCE);
    }

    public RestRequest getRequest() {
        for (PsiElement element : getChildren()) {
            if (element instanceof RestRequest) {
                return (RestRequest) element;
            }
        }
        return null;
    }

    public RestESeparator getSeparator() {
        for (PsiElement element : getChildren()) {
            if (element instanceof RestESeparator) {
                return (RestESeparator) element;
            }
        }
        return null;
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return RestFileType.INSTANCE;
    }
}

File: src/main/kotlin/cn/vove7/plugin/rest/filetype/RestFile.kt
package cn.vove7.plugin.rest.filetype

import com.intellij.psi.FileViewProvider
import com.intellij.extapi.psi.PsiFileBase
import cn.vove7.plugin.rest.filetype.RestLanguage
import cn.vove7.plugin.rest.psi.RestRequest
import com.intellij.psi.PsiElement
import cn.vove7.plugin.rest.psi.RestESeparator
import cn.vove7.plugin.rest.filetype.RestFileType
import com.intellij.openapi.fileTypes.FileType

/**
 *
 */
class RestFile(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, RestLanguage) {
    val request: RestRequest?
        get() {
            for (element in children) {
                if (element is RestRequest) {
                    return element
                }
            }
            return null
        }
    val separator: RestESeparator?
        get() {
            for (element in children) {
                if (element is RestESeparator) {
                    return element
                }
            }
            return null
        }

    override fun getFileType(): FileType {
        return RestFileType.INSTANCE
    }
}